benchmarks/benchmarks.py

```diff
@@ -43,3 +45,16 @@ def time_test_cone_filter_multiple_order():
     assert len(filtered_catalog.partition_info.data_frame) == 2
     assert (6, 30) in filtered_catalog.pixel_tree
     assert (7, 124) in filtered_catalog.pixel_tree
+
+
+class Suite:
+    def setup(self):
+        self.partition_info_df = pd.DataFrame.from_dict(
+            {
+                PartitionInfo.METADATA_ORDER_COLUMN_NAME: np.full(100000, 8),
+                PartitionInfo.METADATA_PIXEL_COLUMN_NAME: np.arange(100000),
+            }
+        )
+
+    def time_pixel_tree_creation(self):
```

Comment:
There's another way to create pixel trees: `from_healpix(healpix_pixels: List[HealpixPixel]) -> PixelTree:`. Was that method sufficiently fast? Was it sped up by these changes?

Reply:
It was sped up a little by the contains improvement, but most of the timesave came from pandas optimizations from the partition_info DF